docs/0.typescript/head/guides/1.core-concepts/5.streaming.md

```
@@ -14,6 +14,50 @@ A streaming server may send its document shell before all head entries exist. Un
+
+```ts
+// the renderer has already run its synchronous pass
+const appStream = renderToWebStream(app)
+return wrapStream(head, appStream, template)
+```
+
+Vue's `renderToWebStream()` runs setup functions while it constructs the stream, but only as far as it can get synchronously. A component's `useHead()` reaches the shell when it runs before that component's own first `await` **and** every ancestor above it is still synchronous at that moment.
+
+That second half is easy to miss. When a component's `setup()` returns a promise, Vue defers rendering its entire subtree, so a child with no `await` of its own still misses the shell:
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)